[inductor] Move ProxyExecutor behind the C shim layer

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #109673
* #109606
* #109450
* __->__ #109498
* #109436

Summary: Move ProxyExecutor behind the C shim layer for ABI compatibility

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

desertfire committed:
[inductor] Move ProxyExecutor behind the C shim layer

Summary: Move ProxyExecutor behind the C shim layer for ABI compatibility

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Move ProxyExecutor behind the C shim layer"


Summary: Move ProxyExecutor behind the C shim layer for ABI compatibility

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

ezyang commented on torch/csrc/inductor/aoti_torch/shim_common.cpp:
```diff
@@ -71,32 +74,28 @@ int32_t aoti_torch_dtype_int64() {
   AOTI_TORCH_CONVERT_EXCEPTION_TO_ERROR_CODE({
-    at::Tensor* t =
-        torch::aot_inductor::tensor_handle_to_tensor_pointer(tensor);
+    at::Tensor* t = tensor_handle_to_tensor_pointer(tensor);
```
This refactor is totally unrelated

desertfire committed:
Update on "[inductor] Move ProxyExecutor behind the C shim layer"


Summary: Move ProxyExecutor behind the C shim layer for ABI compatibility

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Move ProxyExecutor behind the C shim layer"


Summary: Move ProxyExecutor behind the C shim layer for ABI compatibility

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

chenyang78 reviewed (approved):
LGTM. Thanks!

desertfire committed:
Update on "[inductor] Move ProxyExecutor behind the C shim layer"


Summary: Move ProxyExecutor behind the C shim layer for ABI compatibility

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Move ProxyExecutor behind the C shim layer"


Summary: Move ProxyExecutor behind the C shim layer for ABI compatibility

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Move ProxyExecutor behind the C shim layer"


Summary: Move ProxyExecutor behind the C shim layer for ABI compatibility

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

desertfire commented:
Landed with an exported combined PR, https://github.com/pytorch/pytorch/pull/109678
